src/lib-imap/imap-capability-list.c

```diff
@@ -0,0 +1,127 @@
+#include "imap-capability-list.h"
+
+/* initialize the imap capability list */
+struct imap_capability_list *
+imap_capability_list_create(const char *initial_entries)
+{
+	/* initialize a capability list object */
+	struct imap_capability_list *
+		capability_list = i_malloc(sizeof(struct imap_capability_list));
+	capability_list->refcount = 1;
+	/* initialize the capability list memory pool */
+	capability_list->pool = pool_alloconly_create(MEMPOOL_GROWING"Capability List", 2048);
+	/* initialize the capability list capability array */
+	p_array_init(&(capability_list->imap_capabilities),
+			capability_list->pool,
+			0);
+
+	/* make sure our initial entries are appended */
+	if (initial_entries) {
```

Comment:
initial_entries != NULL (we have a patched clang that gives a warning if boolean expressions are mixed with ints or pointers - it has found quite a lot of accidental bugs)